internal/services/kusto/kusto_database_script_resource.go

```diff
@@ -98,8 +98,8 @@ func resourceKustoDatabaseScriptCreateUpdate(d *pluginsdk.ResourceData, meta int
 		}
 	}
 
-	locks.ByID(databaseId.ID())
-	defer locks.UnlockByID(databaseId.ID())
+	locks.ByID(id.ClusterName)
```

Comment:
This needs to be the actual cluster id and not just the name. We can get that id with 
```	
clusterId := parse.NewClusterID(databaseId.SubscriptionId, databaseId.ResourceGroup, databaseId.ClusterName)
locks.ByID(clusterId.ClusterName)
defer locks.UnlockByID(clusterId.ClusterName)
```

Reply:
Updated